src/ray/core_worker/transport/dependency_resolver.cc

```diff
@@ -85,42 +86,49 @@ void LocalDependencyResolver::ResolveDependencies(
     return;
   }
 
-  const auto task_id = task.TaskId();
+  std::weak_ptr<TaskState> weak_task_state;
```

Comment:
why do we need weak_ptr here, we only need task_id in the GetAsync callback lambda?

Reply:
Because we want to decouple task id vs requests. Consider this:

1. request 1: task id 1
2. cancel task id 1 -> cancels all requests for this task id
3. request 2: task id 1
4. mem store callback for request 1 finished
  - we DON'T want to execute that request callback because it's cancelled
5. mem store callback for request 2 finished 
  - we DO want to execute the request callback

so we need to distinguish different reqs with the same task ID. We can do this by an incrementing int ID (RequestID in prev commit), or a weak_ptr to the specific request.

Reply:
per offline discussion, reverted the whole thing of "multiple task id requests". Now it can only accept 1 task id request with CHECKs. Fixed tests to always provide a new task id.